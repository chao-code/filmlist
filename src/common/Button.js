import PropTypes from 'prop-types'
import classNames from 'classnames'
import { Link } from 'react-router-dom'

const Button = props => {
  const { children,
    className,
    href,
    primary,
    secondary,
    danger,
    link,
    small,
    ...attrs
  } = props
  const bsClass = classNames('btn', className, {
    'btn-primary': primary,
    'btn-outline-secondary': secondary,
    'btn-danger': danger,
    'btn-link': link,
    'btn-sm': small,
  })

  return (
    href ? (
      <Link className={bsClass} to={href} {...attrs}>{children}</Link>
    ) : (
      <button className={bsClass} {...attrs}>{children}</button>
    )
  )
}

Button.propTypes = {
  children: PropTypes.string,
  className: PropTypes.string,
  href: PropTypes.string,
  primary: PropTypes.bool,
  secondary: PropTypes.bool,
  danger: PropTypes.bool,
  link: PropTypes.bool,
  small: PropTypes.bool,
}

Button.defaultProps = {
  children: ' ',
  className: '',
  href: '',
  primary: false,
  secondary: false,
  danger: false,
  link: false,
  small: false,
}

export default Button
