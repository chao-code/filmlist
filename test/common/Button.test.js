import { shallow } from 'enzyme'
import Button from '../../src/common/Button'

describe('<Button />', () => {
  it('should render Link', () => {
    const path = '/path/to'
    const wrapper = shallow(<Button href={path}>link</Button>)
    const href = wrapper.find('Link').prop('to')
    expect(href).toBe(path)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render button', () => {
    const onClick = jest.fn()
    const wrapper = shallow(<Button onClick={onClick}>button</Button>)
    wrapper.simulate('click')
    expect(onClick.mock.calls.length).toBe(1)
    expect(wrapper).toMatchSnapshot()
  })
})
