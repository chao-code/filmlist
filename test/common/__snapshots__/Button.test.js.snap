// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render Link 1`] = `
<Link
  className="btn"
  replace={false}
  to="/path/to"
>
  link
</Link>
`;

exports[`<Button /> should render button 1`] = `
<button
  className="btn"
  onClick={[Function]}
>
  button
</button>
`;
